docs/public-api-for-1.0.md

```diff
@@ -0,0 +1,711 @@
+        public static let otlp: Self
+        public static let prometheus: Self
+        public static let console: Self
+    }
+}
+
+extension OTel.Configuration.LogsConfiguration {
+    public struct ExporterSelection: Sendable {
+        public static let otlp: Self
+        public static let console: Self
+    }
+}
+
+extension OTel.Configuration {
+    public struct OTLPExporterConfiguration: Sendable {
+        public var endpoint: String
+        public var insecure: Bool
+        public var certificateFilePath: String?
+        public var clientKeyFilePath: String?
+        public var clientCertificateFilePath: String?
+        public var headers: [(String, String)]
+        public var compression: Compression
+        public var timeout: Duration
+        public var `protocol`: `Protocol`
+
+        public static let `default`: Self
+    }
+}
+
+extension OTel.Configuration.OTLPExporterConfiguration {
+    public struct Compression: Sendable {
+        public static let none: Self
+        public static let gzip: Self
+    }
+
+    public struct `Protocol`: Equatable, Sendable {
+        public static let grpc: Self
+        public static let httpProtobuf: Self
+        public static let httpJSON: Self
+    }
+}
+```
+
+### Reducing visibility of all other existing API
+
+The existing API has biased toward a large public API. This includes all the concrete types that need chaining together
+to construct a functional OTLP backend for each signal, but also a wide API surface for extensibility.
+
+For example, the `OTelTracer` is generic over five abstractions: `OTelIDGenerator`, `OTelSampler`, `OTelPropagator`,
+`OTelSpanProcessor`, and `Clock`, with public concrete types (sometimes several) for each.
+
+Similarly, there is a public abstraction for populating attributes, `OTelResourceDetector`, with several public
+implementations and helper types: `OTelResource`, `OTelResourceDetection`, `OTelEnvironment`,
+`OTelProcessResourceDetector`, `OTelManualResourceDetector`, and `OTelEnvironmentResourceDetector`.
+
+The primary goal of Swift OTel is to provide OTLP backends for the Swift observability APIs and, where appropriate,
+support configuring the behavior according to the OTel specification.
+
+Therefore, we propose to make all the existing API not expressly discussed in the above bootstrap, backends, and
+configuration sections, internal.
+
+Reducing the API surface to just that which is required to meet these goals paves an easier path for a 1.0 release, and
+does not rule out expanding it the future to support deeper customization.
+
+Where extensibility is required, adopters should compose the opaque backends with their own logic. As a concrete
+example, if an adopter wishes to pre-sample traces in a way that is not supported in the OTel specification, they should
+construct their own wrapper type that conforms to `Tracing.Instrument`, which wraps the backend returned by
+`OTel.makeTracingBackend`.
```

Comment:
Perhaps the bootstrap should return `Tracing.Tracer` after all if adopters are expected to wrap it in their own `Tracer` conformance. Otherwise they wouldn't have access to `startSpan` and other Tracing-specific API.